### DISCUSSION TITLE
Trouble Syncing my Plex Servers

### DISCUSSION THREAD

**Discussion by DANW999** [arabcoders/watchstate#371] [General] [answered]
I seem to be having trouble syncing my plex servers, one called plex_main and the other called plex_backup. I have added my two plex servers to watchstate and I have done this several times for my own sanity, completely removing the config and starting again, to make sure all the details are correct. When I then run the state:import command it appears to connected to both servers from the logs, however seemingly on imports it does not import the data from plex_backup, only plex_main. There is a message that says, after it connects to plex_backup "[2023-11-28T17:23:56+00:00] NOTICE: SYSTEM: Waiting on [12] requests." and the proceeds to pull from plex_main but never plex_backup. When I then do state:export, all the connections to plex_backup say "movie Is not imported yet. Possibly because the backend was imported as metadata only." - but I have checked and even reconfigured plex_backup and it is not metadata only. Yet the watch status never updates. I have also run state:requests and there are no requests waiting in the queue. Can I get some help please? I really don't know what is wrong here.

**Comment by arabcoders** [comment]
H could you try importing the problematic backend with these flags and see what you get? e.g. `console state:import -s plex_backup -vvv --trace --context`

**Reply by DANW999** [reply]
Thanks for the prompt response. Just tried that command and its outputting so much, its hard to keep up with the logs. However, I just ran the state:export command again after that and still getting the same issue.

**Reply by arabcoders** [reply]
You need to pipe the results to file and read why it's not importing the backend data it's really impossible to tell without these logs.

**Reply by DANW999** [reply]
OK I've done that. Hard to know what exactly I am looking for though. The beginning reads as so:-

_[2023-11-28T18:00:53+00:00] DEBUG: HttpClient - Request [ HEAD: https://192-168-1-3.805451b1e13a47df8391b6d0c9542107.plex.direct:32400/library/sections/2/all?includeGuids
=1&type=1&sort=addedAt&X-Plex-Container-Start=0&X-Plex-Container-Size=1000] { (options: [ (headers: [ (Accept: application/json), (X-Plex-Token: **hidden**), (X-Plex-Cont
ainer-Size: 1000), (X-Plex-Container-Start: 0) ]), (user_data: [ (library: [ (id: 2), (title: Films), (type: movie), (agent: tv.plex.agents.movie), (url: https://192-168-
1-3.805451b1e13a47df8391b6d0c9542107.plex.direct:32400/library/sections/2/all?includeGuids=1&type=1&sort=addedAt&X-Plex-Container-Start=0&X-Plex-Container-Size=1000) ]), 
(ok: callable), (error: callable) ]) ]) }_

**Reply by arabcoders** [reply]
Thats normal log look for error or something like that and do `console system:report`

You can also add `--logfile /config/import.log` and upload that for me to see whats the problem lies.

**Reply by DANW999** [reply]
The system:report states that task runner failed to run, no PID file. Not sure why though. How can I upload the log file for you?

_[ Basic Report ]

WatchState Version: 20231124-f0e03e2
PHP Version: 8.2.12
Timezone: Europe/London
Running in Container? Yes
Data Path: /config
Temp Path: /config
Database Migrated?: Yes
Has .env file? No
Is Tasks Runner working? No PID file was found - Likely means job-runner failed to run.
Report Generated At: 2023-11-28T18:12:23+00:00

[ Backends ]

[ Plex ==> plex_main ]

Is backend URL HTTPS? Yes
Has Unique Identifier? Yes
Has User? Yes
Export Enabled? Yes
Time since last export? 2023-11-28T17:55:30+00:00
Play state import enabled? Yes
Metadata only import enabled? No
Time since last import? Never
Has webhook token? Yes
Is webhook match user id enabled? No
Is webhook match backend unique id enabled? No
Has custom options? No
{}
[ Plex ==> plex_backup ]

Is backend URL HTTPS? Yes
Has Unique Identifier? Yes
Has User? Yes
Export Enabled? Yes
Time since last export? 2023-11-28T17:55:30+00:00
Play state import enabled? Yes
Metadata only import enabled? No
Time since last import? Never
Has webhook token? Yes
Is webhook match user id enabled? No
Is webhook match backend unique id enabled? No
Has custom options? No
{}
[ Tasks ]

[ Import ]

Is Task enabled? Yes
Which flags are used to run the task? -v
When the task scheduled to run at? */1 * * * *
When is the next scheduled run? 2023-11-28T18:13:00+00:00

[ Export ]

Is Task enabled? Yes
Which flags are used to run the task? -v
When the task scheduled to run at? */1 * * * *
When is the next scheduled run? 2023-11-28T18:13:00+00:00

[ Push ]

Is Task enabled? No

[ Progress ]

Is Task enabled? Yes
Which flags are used to run the task? -v
When the task scheduled to run at? */1 * * * *
When is the next scheduled run? 2023-11-28T18:13:00+00:00

[ Backup ]

Is Task enabled? Yes
Which flags are used to run the task? -v
When the task scheduled to run at? 0 6 */3 * *
When is the next scheduled run? 2023-12-01T06:00:00+00:00

[ Prune ]

Is Task enabled? Yes
Which flags are used to run the task? -v
When the task scheduled to run at? 0 */12 * * *
When is the next scheduled run? 2023-11-29T00:00:00+00:00

[ Indexes ]

Is Task enabled? Yes
Which flags are used to run the task? -v
When the task scheduled to run at? 0 3 * * 3
When is the next scheduled run? 2023-11-29T03:00:00+00:00

[ Requests ]

Is Task enabled? Yes
Which flags are used to run the task? -v --no-stats
When the task scheduled to run at? */2 * * * *
When is the next scheduled run? 2023-11-28T18:14:00+00:00

[ Logs ]

[ /config/logs/app.20231128.log ]

app log file is empty or does not exists.

[ /config/logs/access.20231128.log ]

access log file is empty or does not exists.

[ /config/logs/task.20231128.log ]

[2023-11-28T14:10:55+00:00] WARNING: No backends were found
--------------------------
Task: export (Started: Tue, 14:10:55 GMT)
Command: /opt/app/bin/console state:export -v
Exit Code: 1 (Ended: Tue, 14:10:56 GMT)
--------------------------
No backends were found.

[ /config/logs/app.20231127.log ]

app log file is empty or does not exists.

[ /config/logs/access.20231127.log ]

access log file is empty or does not exists.

[ /config/logs/task.20231127.log ]

task log file is empty or does not exists._

**Reply by arabcoders** [reply]
You can upload it anywhere like https://catbox.moe 

From the report it seems the import never ran so do the following `console state:import -vvv --trace --context --logfile /config/logfull.log` 

upload it and link here

**Reply by DANW999** [reply]
Sorry to be a pain... I'll upload as a last resort... I am just not comfortable with the amount that these logs show about my Plex Servers... I have literally searched for the word error in the logs and the only time it appears is the error: callable messages. Is there anything else I should look our for?

**Reply by arabcoders** [reply]
I can understand not wanting to upload the logs, i need just the first 100 or less lines as usually errors happens there. At least read the messages and it should tell you why.

**Reply by DANW999** [reply]
Please let me know if this is enough? https://litter.catbox.moe/uawdqj.log

**Reply by arabcoders** [reply]
> Please let me know if this is enough? https://litter.catbox.moe/uawdqj.log

says 404 for me sorry i missed the notification. 

If you don't feel like sharing it publicly you could join discord and send it to me there.

**Reply by DANW999** [reply]
Will do. What's the name / link to your discord server please?

**Reply by arabcoders** [reply]
Hi, the discord link is avaliable in the README.md the name there is SupportDude

**Comment by arabcoders** [comment, marked as answer]
After debugging, it was due to user-self imposed `memory_limit` which crashes the import due to the memory hungry import mapper. in memory constraint systems, it's better to use `--direct-mapper` as it uses less memory overall then the defualt mapper.
